courtroom.cpp

```diff
@@ -790,7 +790,10 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
   qDebug() << m_chatmessage[SHOWNAME]
            << ao_app->get_showname(char_list.at(f_char_id).name);
 
-  if (m_chatmessage[SHOWNAME].isEmpty())
+  // We actually DO wanna fail here if the showname is empty but the system is
+  // speaking.
+  // Having an empty showname for system is actually what we expect.
+  if (m_chatmessage[SHOWNAME].isEmpty() && !is_system_speaking)
   {
     f_showname = ao_app->get_showname(char_list.at(f_char_id).name);
   }
```

Comment:
The comment just seems to repeat what we already are looking for inside the if...

Why not just do something like this?:
```cpp
  QString f_showname = m_chatmessage[SHOWNAME];
  if (f_showname.isEmpty() && !is_system_speaking)
    f_showname = ao_app->get_showname(char_list.at(f_char_id).name);

Reply:
The comment explains *why* we're doing it, not *what* we're doing.

And that could also be done, yeah, but it's not like it matters which one we do.
I personally like having curly bracket blocks for each branch, so that in the future, either can be extended as needed without needing to pay much attention as to whether you're still in the if.

Reply:
Then wouldn't something like be more straightforward?
```cpp
  QString f_showname = m_chatmessage[SHOWNAME];
  // Only characters require names.
  if (f_showname.isEmpty() && !is_system_speaking)
  {
    f_showname = ao_app->get_showname(char_list.at(f_char_id).name);
  }